testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java

```diff
@@ -0,0 +1,133 @@
+package org.sonatype.nexus.testsuite.nuget;
+
+import java.util.List;
+import java.util.Map;
+
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.http.HttpStatus;
+
+import org.apache.http.HttpResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Tests NuGet hosted repositories.
+ */
+@ExamReactorStrategy(PerClass.class)
+public class NugetHostedIT
+    extends NugetITSupport
+{
+  private static final String HOSTED_REPO = "nuget-test-hosted";
+
+  private NugetClient nuget;
+
+  @Before
+  public void createHostedRepository() throws Exception {
+    final Configuration config = hostedConfig(HOSTED_REPO);
+
+    final Repository repository = createOrReplaceRepository(config);
+
+    nuget = nugetClient(repository);
+  }
+
+  @After
+  public void deleteHostedRepository() throws Exception {
+    deleteRepository(HOSTED_REPO);
+  }
+
+  @Test
+  public void hostedRepositoryIsAvailable() throws Exception {
+    final String repositoryMetadata = nuget.getRepositoryMetadata();
+    assertThat(repositoryMetadata, is(notNullValue()));
+    assertThat(repositoryMetadata.contains("<Schema Namespace=\"NuGetGallery\""), is(true));
```

Comment:
Usually better expressed using an explicit string matcher, leads to a more meaningful message on error:
```
assertThat(repositoryMetadata, containsString("<Schema Namespace=\"NuGetGallery\""));
```